### ISSUE TITLE
Catalog Graph overwhelmed

### ISSUE THREAD

**Issue by jcarres-mdsol** [backstage/backstage#13260]
We have imported a number of services with their APIs and consumption, and provision information. 
It makes for a very complex graph, some services have lots of consumers (~30s) , others consume a lot of services, etc. 

When selecting the explore graph to render just the consumption and provide relationships, components and APIs, infinite depth we have a graph that makes sense but it is missing relationships.

We have 3 services that are called by almost everything. They should be 3 very clear nodes in the graph, but we can only see one.

<img width="754" alt="Screen Shot 2022-08-19 at 2 22 51 PM" src="https://user-images.githubusercontent.com/1803073/185709023-0bccf5c2-ede8-4331-aac6-2d8bc696d272.png">

That node selected has like 30 incoming lines. Other of the key services (upper right) has a fair number but the third key service has only 5 incoming links, but if I select that one then it will have 30 incoming and the previously selected service just 5.

In short, I think the library is unable or unwilling to draw so many lines and it is removing them for readability, perhaps. 

The problem is that then we have low trust on the graph as we do not know when it will choose to cut corners. 
Also, we can't use it to see our whole platform in one graph that was kind of the goal here.

Settings
<img width="295" alt="Screen Shot 2022-08-19 at 2 29 36 PM" src="https://user-images.githubusercontent.com/1803073/185709792-5437fc08-2814-4ec9-b159-facf72a061e8.png">

**Comment by bendu89** [comment]
Hey - I know that @ZubairRafiq has solved this problem already for one of our clients - I will send him the issue and I'm sure he can help here.

**Comment by ZubairRafiq** [comment]
As you have mentioned that when the maxDepth is set to infinite, perhaps it chooses to cut down several lines. In our case we decided to set the default value of maxDepth to 2 for better readability of graphs. Simply by passing `maxDepth : 2` prop to CatalogGraphPage.

**Comment by freben** [comment]
Maybe worth then, to set the default depth to a smaller value right?

**Comment by ZubairRafiq** [comment]
For sure, that will provide better readability.

**Comment by vorbrodt** [comment]
I would like to work on this issue. Could I please be assigned to it?

**Comment by freben** [comment]
Sure. Assigned!

**Comment by benjdlambert** [comment]
Looks like this was fixed in #14884, closing. Feel free to comment if there's something we missed.

**Comment by jcarres-mdsol** [comment]
I do not think this solves the issue. I could manually look at less information already.
The whole point is to have a view that includes all the components. That view does not render well. That not being the default does not make it render better.

**Comment by Tushti02** [comment]
Hi, The graph issue is still there. Can anybody confirm if this is resolved? 


Also we can see another piece of PR related to this but isn't merged yet

https://github.com/Bonial-International-GmbH/backstage/commit/da0bf25d1a893f1c0ecb19df0431c60f83874781#diff-a4de9e719f3904dc40d526d0f2f529c946f978010111879bda37a5d06b89ff0b

@pjungermann can you confirm on this issue if this was resolved and what actions steps do we need to take.

**Comment by pjungermann** [comment]
@Tushti02 not sure what you mean by "isn't merged yet". It was merged at PR #14466

Also, I think it is not really related to the issue here.
